homeassistant/components/tado/climate.py

```diff
@@ -151,47 +130,51 @@ class TadoClimate(ClimateDevice):
 
     def __init__(
         self,
-        store,
+        tado,
         zone_name,
         zone_id,
-        data_id,
+        zone_type,
         min_temp,
         max_temp,
         step,
-        ac_device,
-        hot_water_device,
         ac_support_heat,
-        tolerance=0.3,
     ):
         """Initialize of Tado climate device."""
-        self._store = store
-        self._data_id = data_id
+        self._tado = tado
 
         self.zone_name = zone_name
         self.zone_id = zone_id
+        self.zone_type = zone_type
 
-        self._ac_device = ac_device
-        self._hot_water_device = hot_water_device
+        self._ac_device = zone_type == TYPE_AIR_CONDITIONING
         self._ac_support_heat = ac_support_heat
         self._cooling = False
 
         self._active = False
         self._device_is_active = False
 
-        self._unit = TEMP_CELSIUS
         self._cur_temp = None
         self._cur_humidity = None
         self._is_away = False
         self._min_temp = min_temp
         self._max_temp = max_temp
         self._step = step
         self._target_temp = None
-        self._tolerance = tolerance
 
         self._current_fan = CONST_MODE_OFF
         self._current_operation = CONST_MODE_SMART_SCHEDULE
         self._overlay_mode = CONST_MODE_SMART_SCHEDULE
 
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+        async_dispatcher_connect(
+            self.hass,
+            SIGNAL_TADO_UPDATE_RECEIVED.format(self.zone_id),
+            self._handle_update,
+        )
+        self._tado.add_sensor("zone", self.zone_id)
+        await self.hass.async_add_executor_job(self._tado.update)
```

Comment:
This is weird. Now it will update Tado, whihc calls `self._handle_update`. But that might be updating before the other entities are initialized. Why can't we update once during component initialization before loading the platforms, and have the entities read the current data in this method ?

Reply:
The previous implementation had a update interval for the component itself, that stored the data, and then a update interval on the platform to fetch that data. This caused a race condition where the platform updated first and got no data, requiring two update intervals to get some data. 

We don't know yet what we need to query during component initialisation, so we initialised the platforms first, then they registers themselves to get updates trough the dispatcher. We need to do a API call per entity (zone) anyway.

I don't know of a better way to do this, for me, this seems pretty clean.

Reply:
There also is a signal per entity, so I don't see how an update could be received before an entity is initialised.

Reply:
entity 1 initializes, triggers update, entity 2 initializes and connects (missed first update), triggers update but hits throttle

Reply:
The first update has no data for entity 2, since it is a different signal, and a different API call. I understand what you mean with the throttling, but I don't see the throttling happening though. Entity 2 gets data just fine. Immediately after startup.

I think that due to the add_executer_job, only one call to update is executed, after the initialisations, at least, that's how I see it behaving locally.

Isn't that what is supposed to happen?

Reply:
I'm wrong, the update happens indeed twice, but I'm lucky and the first update already has the data for the 2nd. I will try to find an alternative way.